backend/app/assets/javascripts/spree/backend/stock_transfer.js.coffee

```diff
@@ -49,8 +49,10 @@ $ ->
       @source.trigger('change')
       if @is_source_location_hidden() and not hide
         $('#transfer_source_location_id_field').css('visibility', 'visible')
+        $('#transfer_source_location_id_field').show()
       else
         $('#transfer_source_location_id_field').css('visibility', 'hidden')
+        $('#transfer_source_location_id_field').hide()
```

Comment:
Yeah it's pretty weird/disgusting. But this is a direct cherry pick so we should probably go with it to minimize the diff from Spree.

https://github.com/spree/spree/blob/master/backend/app/assets/javascripts/spree/backend/stock_transfer.js.coffee#L54